Teapot/Teapot.swift

```diff
@@ -130,8 +130,17 @@ open class Teapot {
                 case .success(let json, let response):
                     // Handle non-2xx status as error.
                     if response.statusCode < 200 || response.statusCode > 299 {
-                        let errorResult = NetworkResult(json, response, TeapotError.invalidResponseStatus(response.statusCode))
-                        completion(errorResult)
+                        // If there is an error sent from backend, its message should be shown
+                        if let errors = json?.dictionary?["errors"] as? [[String: Any]], let error = errors.first, let message = error["message"] as? String {
```

Comment:
This seems like quite the assumption. I think this should be handled by the user. Since we're passing that json along, I don't see how assuming how APIs are designed like this is helpful. If this is something that happens inside a project quite often, the method is `open` and can be override. 👍